codec/src/main/java/io/netty/handler/codec/json/JsonObjectDecoder.java

```diff
@@ -88,6 +90,11 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
             return;
         }
 
+        if (this.idx > in.readerIndex() && lastReaderIndex != in.readerIndex()) {
+            this.idx = in.readerIndex();
+            reset();
```

Comment:
This will reset the state. Is it possible we are in the middle of decoding an object/array when `discardSomeReadBytes` is called? This may move the data around and modify indexes but shouldn't change the data or state. Can you add a unit test covering this case?